sdktests/server_side_events_summary.go

```diff
@@ -259,6 +263,60 @@ func doServerSideSummaryEventPrerequisitesTest(t *ldtest.T) {
 	))
 }
 
+func doServerSideSummaryEventVersionTest(t *ldtest.T) {
+	// This test verifies that if the version of a flag changes within the timespan of one event payload,
+	// evaluations for each version are tracked separately. We do this by evaluating the flag in its
+	// original version, then pushing a stream update and polling until the SDK reports the updated
+	// value, and then checking that both versions appear in the summary event. More detailed testing of
+	// update behavior is covered in server_side_data_store_updates.go.
```

Comment:
Integration-harness sort of covers this case, but its version of the test is less specific and harder to debug, because the summary events for the updated flag are mixed in with the results of all the other evaluations done during the test run.